test/parser/test_base_parser.rb

```diff
@@ -23,5 +23,80 @@ def test_large_xml
         parser.position < xml.bytesize
       end
     end
+
+    def test_parse_exception_for_multiple_root_elements
+      exception = assert_raise(REXML::ParseException) do
+        parser = REXML::Parsers::BaseParser.new('<a></a><b>')
+        while parser.has_next?
+          parser.pull
+        end
+      end
+
+      assert_equal(<<-DETAIL.chomp, exception.to_s)
+Malformed XML: Extra tag at the end of the document (got '<b')
+Line: 1
+Position: 10
+Last 80 unconsumed characters:
+
+      DETAIL
+    end
+
+    def test_parse_exception_for_extra_content_at_the_end_of_the_document
```

Comment:
Could you move this to https://github.com/ruby/rexml/blob/master/test/parse/test_text.rb (the file doesn't exist yet...)?

Reply:
OK.
I see.